### PR TITLE
Terminal Provisioning: generic provision templates

### PR DESCRIPTION
**Pull request by ironBatista**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description
The first part of the provision migration, I've called /provision/v2 in the apache config site. This feature serve a generic template.

### CODE REVIEW

**Comment by ironArt3mis** [comment]
Rebase completed.

**Comment by ironArt3mis** [comment]
Rebase completed.

File: library/Ivoz/Provider/Domain/Model/TerminalModel/TerminalModelRepository.php [hunk 1]
```diff
@@ -11,4 +11,9 @@ interface TerminalModelRepository extends ObjectRepository, Selectable
      * @return TerminalModelInterface | null
      */
     public function findOneByIden(string $iden);
+
+    /**
+     * @return TerminalModelInterface | null
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/TerminalModel/TerminalModelRepository.php:16]: Put this hint in php code findOneByGenericTemplate(string $genericUrlPattern): ?TerminalModelInterface

File: library/Ivoz/Provider/Domain/Model/User/UserRepository.php [hunk 2]
```diff
@@ -53,4 +53,11 @@ public function findCompanyUsersExcludingIds(
     public function findOneByEmail(
         string $email
     );
+
+    /**
+     * @return UserInterface
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/User/UserRepository.php]: Put this hint in php code

File: microservices/provision/src/Controller/ProvisionController.php [hunk 3]
```diff
@@ -1,34 +1,45 @@
...
     use RegisterCommandTrait;
 
     public function __construct(
         DomainEventPublisher $eventPublisher,
-        RequestId $requestId,
-    ) {
+        RequestId            $requestId,
+        private Provision    $provision
+    )
+    {
```
Comment (mmadariaga) [inline microservices/provision/src/Controller/ProvisionController.php]: ) {

File: microservices/provision/src/Services/Provision.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+use Ivoz\Provider\Domain\Model\User\UserRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+class Provision extends AbstractController
+{
+    public function __construct(
+        private TerminalModelRepository $terminalModelRepository,
+        private TerminalRepository $terminalRepository,
+        private UserRepository $userRepository
+    )
```
Comment (mmadariaga) [inline microservices/provision/src/Services/Provision.php]: ) {

File: microservices/provision/src/Services/Provision.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        if($terminalModel){
+            $args = [
+                'terminalModel' => $terminalModel
+            ];
+
+            return $this->renderTemplate( 'generic', $path, $args);
+        }
+
+//        if(!$terminal){
```
Comment (mmadariaga) [inline microservices/provision/src/Services/Provision.php]: Remove comments

File: microservices/provision/src/Services/Provision.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        private UserRepository $userRepository
+    )
+    {
+    }
+
+    public function execute(string $genericUrlPattern, $specificUrlPattern): bool|string
+    {
+        $path = $this->getFilePath();
+        $terminalModel = $this->terminalModelRepository->findOneByGenericTemplate($genericUrlPattern);
+        $terminal = $this->_getTerminalByUrl($specificUrlPattern);
```
Comment (mmadariaga) [inline microservices/provision/src/Services/Provision.php:24]: Unused

File: microservices/provision/src/Services/Provision.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+        return $content;
+    }
+
+    protected function extractFileExtension($route)
+    {
+        return pathinfo($route, PATHINFO_EXTENSION);
+    }
+
+    private function _getTerminalByUrl($terminalUrl)
```
Comment (mmadariaga) [inline microservices/provision/src/Services/Provision.php:74]: unused

File: microservices/provision/templates/basic.txt [hunk 8]
```diff
@@ -0,0 +1,288 @@
+<!DOCTYPE HTML>
```
Comment (mmadariaga) [inline microservices/provision/templates/basic.txt:1]: This is an invoice template, not a provision one

File: microservices/provision/templates/detailed.txt [hunk 9]
```diff
@@ -0,0 +1,314 @@
+<!DOCTYPE HTML>
```
Comment (mmadariaga) [inline microservices/provision/templates/detailed.txt:1]: not provision template

File: microservices/provision/templates/footer.txt [hunk 10]
```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE HTML>
```
Comment (mmadariaga) [inline microservices/provision/templates/footer.txt:1]: Not provision template

File: microservices/provision/templates/header.txt [hunk 11]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE HTML>
```
Comment (mmadariaga) [inline microservices/provision/templates/header.txt:1]: Not provision temnplate

**Review by mmadariaga — CHANGES_REQUESTED** [review]

**Comment by ironArt3mis** [comment]
Rebase completed.

File: microservices/provision/src/Services/Provision.php [hunk 12]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Services;
+
+use Ivoz\Provider\Domain\Model\TerminalModel\TerminalModelInterface;
+use Ivoz\Provider\Domain\Model\TerminalModel\TerminalModelRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+/** @psalm-suppress PropertyNotSetInConstructor */
+class Provision extends AbstractController
```
Comment (mmadariaga) [inline microservices/provision/src/Services/Provision.php:10]: Don't extend AbstractController, just register controllers as a service

**Review by mmadariaga — APPROVED** [review]

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**